src/SubscriptionModel.m

```diff
@@ -57,18 +57,19 @@ -(NSURL *)verifiedFeedURLFromURL:(NSURL *)rssFeedURL
 	// That would require modifying extractFeeds to provide URL strings and titles
 	// as feeds' links are often advertised in the HTML head
 	// as <link rel="alternate" type="application/rss+xml" title="..." href="...">
+	NSURL * myURL;
     if ([RichXMLParser extractFeeds:urlContent toArray:linkArray])
     {
         NSString * feedPart = linkArray.firstObject;
-        if (![feedPart hasPrefix:@"http:"] && ![feedPart hasPrefix:@"https:"])
-        {
-			rssFeedURL = [NSURL URLWithString:feedPart relativeToURL:rssFeedURL];
-        }
-        else {
-            rssFeedURL = [NSURL URLWithString:feedPart];
-        }
+		myURL = [NSURL URLWithString:feedPart relativeToURL:rssFeedURL];
+		if (myURL ==nil)
+		{
+			// try cleaning up the string : unescape then re-add escapes
+			NSString * urlString = [[feedPart stringByUnescapingExtendedCharacters] stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+			myURL = [NSURL URLWithString:urlString relativeToURL:rssFeedURL];
+		}
     }
-    return rssFeedURL.absoluteURL;
+    return myURL;
```

Comment:
This change seems to be what is causing the "new" issue. If `myURL` isn't set because `extractFeeds` returns `NO`, then we get `nil` returned even if the original URL was OK.